docs/content/controls/README.md

```diff
@@ -0,0 +1,185 @@
+  abstract public function content_template();
+
+  // The control type. Each control has to implement this method.
+  abstract public function get_type();
+
+  // The control constructor. Each control may override this method or leave its default functionality.
+  public function __construct() {}
+
+  // If the control need an additional assets libraries (such as JS, CSS etc.),
+  // It will be enqueued here.
+  public function enqueue() {}
+
+  // Determines the default value that the control will return.
+  public function get_default_value() {}
+
+  // Determines how the control returns its value.
+  // This method gets a control instance settings and widget instance settings
+  // and decides the value will be returned.
+  // Each control may override this method or leave its default functionality.
+  public function get_value( $control, $widget ) {}
+
+  // Determines how the control returns a style value. (See values types below).
+  public function get_style_value( $css_property, $control_value ) {}
+
+  // Get whole settings of the control. This method could not be overrided.
+  final public function get_settings( $setting_key = null ) {}
+
+  // Prints the control template. This method could not be overrided.
+  final public function print_template() {}
+
+  // Return the specific control type additional settings.
+  // Each control may override this method or leave its default (empty).
+  protected function get_default_settings() {}
+
+}
+```
+
+###Control settings
+
+####Default settings
+Each control has a set of default settings that determine the structure and design of the control, that are displayed while the widget is being edited.
+
+Here is the list of control settings that all control have in common:
+
+* `label` - The title of the control
+
+* `default` - The default value
+
+* `separator` - Sets the position of the separator: `'before'` / `'default'` / `'after'` / `'none'`. *Default*: `'default'`
+
+* `label_block` - Sets whether to display the title in a separate line: `true` / `false`. *Default*: `false`
+
+* `show_label` - sets whether to show the title: `true` / `false`. *Default*: `true`
+
+* `title` - The title that will appear on mouse over
+
+* `placeholder` - Available for fields that support placeholder
+
+* `description` - A description of the field, that appears below the field
+
+####Settings Hierarchy
+In addition to the default settings that come with the basic control, each control can be added with its own default settings, and can also have settings that override the default class.
+
+In addition, with every addition of a control to a widget, you can send another set of settings that will override the entire default settings. Here is an example:
+
+Default settings in the base class:
+
+```php
+abstract class Control_Base {
+
+  // here is some base default settings
+  private $_base_settings = [
+     'separator' => 'default',
+     'label' => ‘’,
```

Comment:
Maybe it's a good idea to search for `‘` and `’` (apostrophes?) and replace them with `'` (single quote)?